code/modules/mob/living/simple_animal/abnormality/waw/sphinx.dm

```diff
@@ -0,0 +1,469 @@
+				QuestPenalty(user)
+				return
+
+/mob/living/simple_animal/hostile/abnormality/sphinx/proc/QuestReward()
+	say(pick(satisfied))
+	happy = TRUE
+	demand = 0
+	datum_reference.qliphoth_change(3)
+	var/turf/dispense_turf = get_step(src, pick(1,2,4,5,6,8,9,10))
+	var/reward = pick(riddleloot)
+	new reward(dispense_turf)
+
+/mob/living/simple_animal/hostile/abnormality/sphinx/proc/QuestPenalty(mob/living/carbon/human/user)
+	say(pick(angry))
+	datum_reference.qliphoth_change(-1)
+	var/mob/living/carbon/human/H = user
+	var/obj/item/organ/eyes/eyes = H.getorganslot(ORGAN_SLOT_EYES)
+	var/obj/item/organ/ears/ears = H.getorganslot(ORGAN_SLOT_EARS)
+	var/obj/item/organ/tongue/tongue = H.getorganslot(ORGAN_SLOT_TONGUE)
+	var/chosenorgan = pick(eyes,ears,tongue)
+	while(!chosenorgan)
+		if(!eyes && !ears && !tongue)
+			to_chat(H, "<span class='warning'>With nothing left to lose, you lose your life. </span>")
+			H.dust()
+			return
+		chosenorgan = pick(eyes,ears,tongue)
+	if(chosenorgan == eyes)
+		to_chat(user, "<span class='warning'>A brillaint flash of light is the last thing you see...</span>")
+	if(chosenorgan == ears)
+		to_chat(user, "<span class='warning'>Suddenly, everything goes quiet...</span>")
+	if(chosenorgan == tongue)
+		to_chat(user, "<span class='warning'>Your mouth feels uncomfortably hollow...</span>")
+	H.internal_organs -= chosenorgan
+	qdel(chosenorgan)
+
+//Breach
+/mob/living/simple_animal/hostile/abnormality/sphinx/BreachEffect(mob/living/carbon/human/user)
+	..()
+	AddComponent(/datum/component/knockback, 3, FALSE, TRUE)
+	GiveTarget(user)
+
+/mob/living/simple_animal/hostile/abnormality/sphinx/Move()
+	if(!can_act)
+		return FALSE
+	..()
+
+/mob/living/simple_animal/hostile/abnormality/sphinx/AttackingTarget()
+	..()
+	if(ishuman(target))
+		var/mob/living/carbon/human/H = target
+		if(H.sanity_lost)
+			QDEL_NULL(H.ai_controller)
+			H.ai_controller = /datum/ai_controller/insane/wander/sphinx
+			H.InitializeAIController()
+			LoseTarget(H)
+
+/mob/living/simple_animal/hostile/abnormality/sphinx/OpenFire()
+	if(!can_act)
+		return
+
+	if((curse_cooldown <= world.time))
+		StoneVision()
+	return
+
+/mob/living/simple_animal/hostile/abnormality/sphinx/proc/StoneVision()
+	if((curse_cooldown > world.time) && !attack_chain)
+		return
+	if(!attack_chain)
+		playsound(get_turf(src), 'sound/abnormalities/sphinx/stone_ready.ogg', 50, 0, 5)
+		for(var/turf/T in view(7, src))
+			if(T.density)
+				continue
+			new /obj/effect/temp_visual/stone_gaze(T)
+	curse_cooldown = world.time + curse_cooldown_time
+	icon_state = icon_aggro
+	can_act = FALSE
+	new /obj/effect/temp_visual/eyewarning(get_turf(src))
+	SLEEP_CHECK_DEATH(12)
+	for(var/mob/living/carbon/human/L in viewers(7, src))
+		if(L.client && CanAttack(L) && L.stat != DEAD)
+			if(!L.is_blind() && is_A_facing_B(L,src))
+				StoneCurse(L)
+	if(!attack_chain)
+		attack_chain = TRUE
+		StoneVision()
+	else
+		attack_chain = FALSE
+	icon_state = icon_living
+	can_act = TRUE
```

Comment:
There's no delay between this and the next icon_state change and the next can_act change, I'm worried this could result in it moving when it shouldn't or otherwise failing to change icon properly